src/vuex/modules/keystore.js

```diff
@@ -73,29 +73,35 @@ export default () => {
 }
 
 // creates a cosmos addres for the network desired
-function getCosmosAddressCreator(network) {
+function getCosmosAddressCreator(network, networkPrefix) {
   return async seedPhrase => {
     const { getNewWalletFromSeed } = await import("@lunie/cosmos-keys")
-    return getNewWalletFromSeed(seedPhrase, config.bech32Prefixes[network])
+    networkPrefix = networkPrefix || config.bech32Prefixes[network]
+    return getNewWalletFromSeed(seedPhrase, networkPrefix)
   }
 }
 
-async function getWallet(seedPhrase, network) {
-  switch (network) {
-    case "cosmos-hub-mainnet":
-    case "cosmos-hub-testnet":
-    case "regen-testnet":
-    case "regen-mainnet":
-    case "terra-testnet":
-    case "emoney-testnet":
-    case "emoney-mainnet":
-    case "terra-mainnet": {
-      const addressCreator = await getCosmosAddressCreator(network)
-      return addressCreator(seedPhrase)
+async function getWallet(seedPhrase, network, networkPrefix) {
+  if (networkPrefix) {
```

Comment:
can you explain what this is about?

Reply:
this is about sending the right prefix to cosmos-keys to create the address

Reply:
but why are you moving around the network check in line 89?

Reply:
Ah ok. My thought here was like, if we are calling the function without passing through selectnetwork, and therefore we don't get the networkPrefix parameter, then do it this other way just by using the network (the logic we had before).

But as you have made clear, the user will always go through the SelectNetwork component in the flow of creating an account. So we can just delete those lines and keep networkPrefix as the second parameter and delete network.

Did I answer your question?

Reply:
checkout my version here: https://github.com/luniehq/lunie/pull/3456/files#diff-5bdea6c484046b75be42e1cf57c1167bR81
I expect the same flow for selecting the network. It will not pass the prefix around. It will ONLY pass the network ID around. The function that needs more information will then query for the network information like the prefix it needs.

Reply:
It just refactored it and it is working now like in your PR :wink: